crates/bevy_ecs/src/system/system.rs

```diff
@@ -390,6 +389,13 @@ impl RunSystemOnce for &mut World {
     }
 }
 
+/// Allows for systems to inform their names, either derived from their type
+/// or their custom name.
+pub trait NamedSystem {
```

Comment:
I don't understand why this trait is necessary: the `System` trait already has a `name` method?